### PR TITLE
Initial setup process and related features

### LINKED ISSUE #308: 非UTF8環境でHostName(等)がASCII範囲外を含む場合に警告する

### LINKED ISSUE #593: 初回セットアップの仕組み
#308 とか #268 の追加や #391 とかであるべき

search terms: setup initial

初回セットアップでやるべきこと
- Locale Settings
- Color Theme
- install unity hub if not exists
- install unity if not exists
- backup format
- default project / backup 場所の更新
  - if VCC is not configured yet, I think it's better to save at `Documents/VRChatProjects` and `Documents/VRChatBackups`
- ask for installing `vcc:` handler 

Related
- #308
- #268
- #391
- #612

### LINKED ISSUE #612: デフォルトのプロジェクト作成先を変える
### Summary / 概要

デフォルトのプロジェクト作成先を変える

### Purpose / 目的

vrchat-community/creator-companion#319

"reset PC but keep personal files" で消えるらしいので

### Additional context / 補足

_No response_

### PR DESCRIPTION
**Pull request by anatawa12**
Fixes #593 
Fixes #308 
Fixes #612 

TODOs:
- [x] save if setup finished
- [x] show progress
- [x] localize
- [x] ~~height change animation~~ impossible. Implemented shared height instead

### CODE REVIEW

File: vrc-get-gui/locales/en.json5 [hunk 1]
```diff
@@ -6,6 +6,9 @@
     "vpm repositories": "VPM Repositories",
     "logs": "Logs",
     "sidebar:toast:version copied": "Version name was copied.",
+    "sidebar:bad hostname": "Bad hostname",
+    "sidebar:dialog:bad hostname": "The hostname is not good.",
```
Comment (anatawa12) [inline vrc-get-gui/locales/en.json5:10]:
```suggestion
    "sidebar:dialog:bad hostname": "Your hostname is not good.",
```

Comment (JustBuddy) [reply vrc-get-gui/locales/en.json5:10]: Odd wording, you should use "Entered hostname is invalid".

Comment (anatawa12) [reply vrc-get-gui/locales/en.json5:10]: the hostname is already configured and not entered in ALCOM so "Entered" is odd I think. 
In addition, hostname with non ASCII is not invalid but may not be incompatible with unity so not good. (bad?)

"Your hostname (PC name) is not good" or "Your PC hostname is bad"?

Comment (JustBuddy) [reply vrc-get-gui/locales/en.json5:10]: Oh, I wasn't 100% clear about the context of this key.
Maybe "There are issues with your hostname." or something like that.
Something that isn't "is not good", and "bad" sounds judgmental.

Comment (anatawa12) [reply vrc-get-gui/locales/en.json5:10]: Thank you for suggestion!

```suggestion
    "sidebar:dialog:bad hostname": "There are issues with your hostname.",
```

File: vrc-get-gui/app/(setup)/setup/setup-page-base.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+) {
+	const router = useRouter();
+
+	const result = useQuery({
+		queryKey: ["environmentGetSettings"],
+		queryFn: environmentGetSettings,
+	})
+
+	const onBack = () => {
+		router.back()
```
Comment (anatawa12) [inline vrc-get-gui/app/(setup)/setup/setup-page-base.tsx:46]: 途中から始めたときに困る

**Comment by anatawa12** [comment]
Current UI

image
image
image
image
image
image

**Comment by JustBuddy** [comment]
@anatawa12 `setup:finish:heading` doesn't seem to be applied correctly, the header doesn't change in translation.

**Comment by anatawa12** [comment]
@JustBuddy Thank you for notice that. I'll fix in next PR.

File: vrc-get-gui/app/(setup)/setup/unity-hub/page.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+					refetch={refetch}
+					notFoundMessage={"Unity Hub Not Found"}
+					successMessage={tc("settings:toast:unity hub path updated")}
+				/>
+			</>
+			: <>
+				<div className={"p-2"}/>
+				<div className={"flex flex-row flex-wrap gap-2"}>
+					<Button onClick={() => shellOpen("https://unity.com/ja/download")}>
+						{tc("Download Unity Hub from unity.com")}
```
Comment (Sayamame-beans) [inline vrc-get-gui/app/(setup)/setup/unity-hub/page.tsx:49]: キーが違います

File: vrc-get-gui/locales/en.json5 [hunk 4]
```diff
@@ -274,5 +278,44 @@
...
+    "setup:unity-hub:description": "To get started, you need to install Unity Hub. Unity Hub is the preferred way to install and manage Unity versions.<br/>ALCOM will suggest you installing Unity with Unity Hub if you don't have and necessary for the project or operation.",
+    "setup:unity-hub:using this unity hub": "Using this Unity Hub",
+    "setup:unity-hub:download unity hub from unity.com": "Download Unity Hub from unity.com",
+    "setup:unity-hub:recheck installation": "Recheck Installation",
+    "setup:unity-hub:detection failed collapse": "Have you installed Unity Hub?",
+    "setup:unity-hub:detection failed description": "Please click the above \"Recheck Installation\" button to re-detect Unity Hub installation. If this doesn't work, ALCOM fails to detect your Unity Hub installation. Please set Unity Hub Manually below.",
+    "setup:unity-hub:not found": "Unity Hub was not found.",
+    "setup:project-path:heading": "Project Save Location",
+    "setup:project-path:description": "When you crete project with ALCOM, the project will be saved in this location.<br/>This setting may also be changed in settings page later.",
```
Comment (Sayamame-beans) [inline vrc-get-gui/locales/en.json5:301]: メモ
```diff
- When you crete project with ALCOM,
+ When you create project with ALCOM,
```

Comment (Sayamame-beans) [reply vrc-get-gui/locales/en.json5:301]: > This setting may also be changed in settings page later.

後から変更出来ない設定ってあります…?

Comment (anatawa12) [reply vrc-get-gui/locales/en.json5:301]: ないつもり

Comment (Sayamame-beans) [reply vrc-get-gui/locales/en.json5:301]: であれば、個別に書くよりどこかに固定で載せておいた方が良いような気もしたり…?
個別に書くと、書いてないところはそうじゃないように見えますし、かといって全部書くとしつこいと思うので

File: vrc-get-gui/locales/en.json5 [hunk 5]
```diff
@@ -274,5 +278,44 @@
...
+    "setup:project-path:warning:whitespace": "The location contains whitespace. Whitespace in may cause problems with Unity and other tools. Please consider use another location.",
+    "setup:project-path:warning:non-ascii": "The location contains non-ASCII characters. Non-ASCII characters in the location locale may cause problems with Unity. Please consider use another location.",
```
Comment (Sayamame-beans) [inline vrc-get-gui/locales/en.json5:303]: "location locale"って誤字です…?

Comment (anatawa12) [reply vrc-get-gui/locales/en.json5:303]: non UTF8 localeって話を一旦入れてて冗長かって消したんだけどその消し忘れっぽい

File: vrc-get-gui/locales/en.json5 [hunk 6]
```diff
@@ -274,5 +278,44 @@
...
+    // Reset your PC is a feature in Windows, so please check your translation is consistent with Windows terminology.
+    // It's at Setting app > System > Recovery > Reset this PC
+    "setup:project-path:warning:in-local-app-data": "The location is in LocalAppData folder, which will be deleted with \"Reset this PC\" with \"Keep my files\".\nIt's recommended to save your projects in a different location.",
```
Comment (Sayamame-beans) [inline vrc-get-gui/locales/en.json5:306]: \nだと改行出来てなさそう

File: vrc-get-gui/locales/en.json5 [hunk 7]
```diff
@@ -274,5 +278,44 @@
...
+    "setup:backups:heading": "Project Backup Settings",
+    "setup:backups:location": "Backup Location",
+    "setup:backups:location description": "When you backup your project, the backup archive file will be saved in this path.<br/>This setting may also be changed in settings page later.",
+    "setup:backups:warning:in-local-app-data": "The location is in LocalAppData folder, which will be deleted with \"Reset this PC\" with \"Keep my files\".\nIt's recommended to save your backups in a different location.",
+    "setup:backups:archive": "Archive Format",
+    "setup:backups:archive description": "Backups are saved as an archive file. <br/>You can select the format of the backup archive. <br/>Default setting is the best format ALCOM thinks at the moment and current default behavior is represented in braces.<br/>Compressing the archive will make backup took longer but will save disk space.",
```
Comment (Sayamame-beans) [inline vrc-get-gui/locales/en.json5:312]: この辺り、設定ページにない警告や説明分が初期設定時にだけあるのに少し違和感あり…?

**Pull request merged**